### ISSUE TITLE
[Code health] Consolidate string resources

### ISSUE THREAD

**Issue by gino-m** [google/ground-android#334]
From @shobhitagarwal1612:
> I also noticed that some of strings are duplicate (like add_feature_cancel and discard_multiple_choice_changes). Any particular reason for that? Also, would it be fine to use android's resource for common strings (android.R.string.cancel) ?
https://github.com/google/ground-android/pull/332#issue-367091582

I don't remember why we have these duplicate strings - consolidating and using default strings sounds good.

**Comment by gino-m** [comment]
@dturner  is there any official guidance on whether to merge identical UI strings or not?

We didn't proceed with the linked PR because in some case we have what appear to be duplicate strings for different actions in different UI surfaces. I assuemd splitting strings semantically rather than literally is important because the same phrases may be worded differently in different languages. It also allows us to easily change the label for one UI element without changing others (e.g., "Add" => "Add feature" and "Add observation). Any thoughts on this?

**Comment by dturner** [comment]
Taking a look at the [string resources in the IOsched app](https://github.com/google/iosched/blob/master/mobile/src/main/res/values/strings.xml) (which attempts to follow best practice) it looks like the following convention is used: 

- `name` 2-4 word semantic description of the string
- `translation_description` describes what the string is used to communicate, where it is presented and any events which result in it being displayed. This presumably makes it easy for a translator.

It looks like you already have the text for `translation_description` in a comment above each string resource.

As for identical strings, I don't think we have any specific guidance. My gut feel is that having a single instance of `<string name=cancel>Cancel</string>` is better than having `<string name=cancel_when_event1>Cancel</string>` and `<string name=cancel_when_event2>Cancel</string>` because it makes the code more readable. i.e.

    dialogBuilder.setNegativeButton(R.string.cancel, (dialog, which) -> {});

is more readable than

    dialogBuilder.setNegativeButton(R.string.discard_multiple_choice_changes, (dialog, which) -> {});

In summary, I think removing the duplicates makes sense.

**Comment by gino-m** [comment]
Ok, that sounds reasonable. Thanks for the input!

@shobhitagarwal1612 since you already did this, case to patch #346 and send a PR?

**Comment by shobhitagarwal1612** [comment]
Sure. Thanks for the guidance.
